src/dsp_tools/utils/project_create.py

```diff
@@ -6,72 +6,142 @@
+        print('\tJSON project file is syntactically correct and passed validation.')
+    except BaseError as err:
+        print(f'=====================================\n'
+              f'{err.message}')
+        quit(0)
+
+    # establish connection to DSP server
+    con = login(server=server, user=user_mail, password=password)
+    if dump:
+        con.start_logging()
 
-def _update_project(project: Project, project_definition: dict[str, Any], verbose: bool) -> Project:
+    # if project exists, update it, otherwise create it
+    try:
+        project_local = Project(con=con, shortcode=project_definition["project"]["shortcode"])
+        project_remote: Project = try_network_action(
+            action=lambda: project_local.read(),
+            failure_msg=""
+        )
+        print(f"\tWARNING: Project '{project_remote.shortname}' ({project_remote.shortcode}) already exists on the DSP "
+              f"server. Updating it...")
+        overall_success = False
+        project_remote, _ = _update_project(project=project_remote, project_definition=project_definition, verbose=verbose)
+    except BaseError:
+        project_local = Project(
+            con=con,
+            shortcode=project_definition["project"]["shortcode"],
+            shortname=project_definition["project"]["shortname"],
+            longname=project_definition["project"]["longname"],
+            description=LangString(project_definition["project"].get("descriptions")),
+            keywords=set(project_definition["project"].get("keywords")),
+            selfjoin=False,
+            status=True
+        )
+        project_remote = try_network_action(
+            action=lambda: project_local.create(),
+            failure_msg=f"ERROR: Cannot create project '{project_definition['project']['shortname']}' "
+                        f"({project_definition['project']['shortcode']}) on DSP server."
+        )
+        print(f"\tCreated project '{project_remote.shortname}' ({project_remote.shortcode}).")
+
+    return con, context, project_definition, project_remote, overall_success
+
+
+def _update_project(
+        project: Project,
+        project_definition: dict[str, Any],
+        verbose: bool
+) -> Tuple[Project, bool]:
     """
     Updates a project on a DSP server from a JSON project file. Only the longname, description and keywords will be
-    updated. Raises a BaseError if the project cannot be updated.
+    updated. Returns the updated project (or the unchanged project if not successful) and a boolean saying if the update
+    was successful or not. If the update was not successful, an error message is printed to stdout.
 
     Args:
         project: the project to be updated (must exist on the DSP server)
         project_definition: a parsed JSON project file with the same shortname and shortcode than the existing project
 
     Returns:
-        updated project
+        tuple of (updated project/None, success status)
```

Comment:
in what case would project be `None`? Also, the return type does not reflect that

Reply:
That's a leftover that I forgot to tidy up. Thanks for spotting it